### PR TITLE
[Heartbeat] Zip URL Support

### PR DESCRIPTION
**Pull request by andrewvc**
Adds support for remote zips to browser monitors, letting users just automatically pull from a remote repo.

Fixes https://github.com/elastic/uptime/issues/271

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/heartbeat/monitors/browser/source/zipurl.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+	rdr, err := zip.NewReader(tf, stat.Size())
+	if err != nil {
+		return fmt.Errorf("could not read file %s as zip: %w", tf.Name(), err)
+	}
+
+	for _, f := range rdr.File {
+		err = unzipFile(targetDir, folder, f)
+		if err != nil {
+			// TODO: err handlers
+			os.RemoveAll(targetDir)
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/source/zipurl.go:102]: This feels unnecessary right? we already remove the directory when unzip returns error. 

https://github.com/elastic/beats/pull/24714/files#diff-a42b602f839672a041979a73130887f65c77d4792a207ce86444af8a01660012R71

File: x-pack/heartbeat/monitors/browser/source/zipurl_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source/fixtures"
+)
+
+func TestZipUrlFetchNoAuth(t *testing.T) {
+	address, teardown := setupTests()
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/source/zipurl_test.go:23]: I thought doing this for every test was a overkill which is why i moved up everything in to functions and called them from a global test function. I guess it's not?  Just a question though.

**Review by vigneshshanmugam — APPROVED** [review]
Looks good to me 👍

**Comment by andrewkroh** [comment]
@andrewvc One of the new tests in the PR is panicking in master.

 https://beats-ci.elastic.co/job/Beats/job/beats/job/master/560/testReport/junit/github/com_elastic_beats_v7_x-pack_heartbeat_monitors_browser_source/Build_Test___x_pack_heartbeat_build___TestZipUrlTargetDirectory/


```
=== RUN   TestZipUrlTargetDirectory
--- FAIL: TestZipUrlTargetDirectory (1.50s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x73bd49]

goroutine 70 [running]:
testing.tRunner.func1.1(0x785400, 0xa520e0)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/.gvm/versions/go1.15.9.linux.amd64/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc0002a2480)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/.gvm/versions/go1.15.9.linux.amd64/src/testing/testing.go:1075 +0x41a
panic(0x785400, 0xa520e0)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/.gvm/versions/go1.15.9.linux.amd64/src/runtime/panic.go:969 +0x1b9
github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source.retryingZipRequest(0x7e9d3d, 0x4, 0xc0000e1ef0, 0xe0, 0xd8, 0x7de820)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/src/github.com/elastic/beats/x-pack/heartbeat/monitors/browser/source/zipurl.go:177 +0x189
github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source.checkIfChanged(0xc0000e1ef0, 0x40f100, 0x0, 0x0)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/src/github.com/elastic/beats/x-pack/heartbeat/monitors/browser/source/zipurl.go:212 +0x6e
github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source.(*ZipURLSource).Fetch(0xc0000e1ef0, 0x0, 0x0)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/src/github.com/elastic/beats/x-pack/heartbeat/monitors/browser/source/zipurl.go:34 +0x73
github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source.fetchAndCheckDir(0xc0002a2480, 0xc0000e1ef0)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/src/github.com/elastic/beats/x-pack/heartbeat/monitors/browser/source/zipurl_test.go:118 +0x3f
github.com/elastic/beats/v7/x-pack/heartbeat/monitors/browser/source.TestZipUrlTargetDirectory(0xc0002a2480)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/src/github.com/elastic/beats/x-pack/heartbeat/monitors/browser/source/zipurl_test.go:57 +0x14c
testing.tRunner(0xc0002a2480, 0x805e78)
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/.gvm/versions/go1.15.9.linux.amd64/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/var/lib/jenkins/workspace/master-560-11f31d70-23ff-4709-a29c-a54165d77063/.gvm/versions/go1.15.9.linux.amd64/src/testing/testing.go:1168 +0x2b3
```

**Comment by andrewvc** [comment]
@andrewkroh fixed it in https://github.com/elastic/beats/pull/25288 , apologies for the flakiness!

**Pull request merged**